common/oauthTokenManager.go

```diff
@@ -771,13 +770,41 @@ func (credInfo *OAuthTokenInfo) GetTokenCredential() (azcore.TokenCredential, er
 	}
 }
 
+// MarshalJSON customizes the JSON encoding for AutoLoginTypeJSON
+func (alt AutoLoginTypeJSON) MarshalJSON() ([]byte, error) {
+	return json.Marshal(alt.AutoLoginType)
+}
+
+// UnmarshalJSON customizes the JSON decoding for AutoLoginTypeJSON
+func (alt *AutoLoginTypeJSON) UnmarshalJSON(data []byte) error {
+	var v interface{}
+	if err := json.Unmarshal(data, &v); err != nil {
+		return err
+	}
+
+	switch value := v.(type) {
+	case float64:
+		alt.AutoLoginType = AutoLoginType(value)
+	case string:
+		uintValue, err := strconv.ParseUint(value, 10, 8)
```

Comment:
I think we may have miscommunicated a bit. This isn't the intended behavior-- `AutoLoginType` is an enum, if you check the type's `String` and `Parse` functions, this is what we're looking for, regarding a string value here.

Reply:
updated the code accordingly